c/zis/client.c

```diff
@@ -162,25 +164,51 @@ int zisGenerateOrValidateSafIdt(const CrossMemoryServerName *serverName,
   }
   strncpy(parmList.passwordNullTerm, password, sizeof(parmList.passwordNullTerm));
 
-  parmList.options |= ZIS_AUTH_SERVICE_PARMLIST_OPTION_GENERATE_IDT;
-  parmList.safIdtLen = strlen(safIdt);
+  if (appl != NULL && strlen(appl) != 0) {
+    if (strlen(appl) >= sizeof (parmList.safIdtService.applNullTerm)) {
+      status->baseStatus.serviceRC = RC_ZIS_AUTHSRV_INPUT_STRING_TOO_LONG;
+      return RC_ZIS_SRVC_SERVICE_FAILED;
+    }
+    strcpy(parmList.safIdtService.applNullTerm, appl);
+
+    parmList.safIdtService.options |= ZIS_AUTH_SERVICE_SAFIDT_OPTION_IDT_APPL;
+  }
 
-  if (strlen(safIdt) >= sizeof(parmList.safIdt)) {
+  parmList.safIdtService.safIdtLen = strlen(safIdt);
+  parmList.safIdtService.safIdtServiceVersion = 1;
+  if (strlen(safIdt) >= sizeof(parmList.safIdtService.safIdt)) {
     status->baseStatus.serviceRC = RC_ZIS_AUTHSRV_INPUT_STRING_TOO_LONG;
     return RC_ZIS_SRVC_SERVICE_FAILED;
   }
-  memcpy((void *)parmList.safIdt, (void *)safIdt, strlen(safIdt));
+  memcpy((void *)parmList.safIdtService.safIdt, (void *)safIdt, strlen(safIdt));
 
   int rc = authRequest(serverName, &parmList, status);
 
-  if (parmList.safIdtLen > 0) {
-    memset((void *)safIdt, 0, ZIS_AUTH_SERVICE_PARMLIST_SAFIDT_LENGTH + 1);
-    memcpy((void *)safIdt, (void *)parmList.safIdt, parmList.safIdtLen);
+  if (parmList.safIdtService.safIdtLen > 0) {
```

Comment:
Help me understand the logic. Why do we need this block of code? If SAF doesn't return anything to us except for the return codes, can we not do this? If so, can we just make `safIdt` an input parameter only and mark it as `const`?

Reply:
N/A